client/components/FacetsStateProvider/state.ts

```diff
@@ -0,0 +1,56 @@
+import {
+	FacetCascadeResult,
+	FacetDefinition,
+	FacetSourceScope,
+	CascadedFacetsByKind,
+	mapFacetDefinitions,
+} from 'facets';
+import { Scope } from 'types';
+import { FacetsState, FacetState } from './types';
+
+export type CreateStateOptions = {
+	currentScope: FacetSourceScope | Scope;
+	cascadeResults: CascadedFacetsByKind;
+};
+
+function createInitialFacetState<Def extends FacetDefinition>(
+	facetDefinition: Def,
+	initialCascadeResult: FacetCascadeResult<Def>,
+): FacetState<Def> {
+	return {
+		facetDefinition,
+		persistedCascadeResult: initialCascadeResult,
+		cascadeResult: initialCascadeResult,
+		latestAndPossiblyInvalidCascadeResult: initialCascadeResult,
+		persistableChanges: {},
+		invalidProps: {},
+		hasInvalidChanges: true,
```

Comment:
could this be null or true? I don't understand having an initial state of `true` for a property that should be changed by an update

Reply:
That's because this is supposed to be `false` 😅